### PR TITLE
Added keys to tags in <Head> component

### PR DESCRIPTION
**Pull request by PierreRe**
Added keys to to tags in Next.JS <Head> component to prevent duplication in rendered HTML when rewriting meta-tags in child components as best-practice approach.

### CODE REVIEW

**Comment by JohnAlbin** [comment]
From my understanding `key` is only needed for each item in an array. https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key

Next.js's `Head` docs show using a `key` when `Head` is used multiple times on a single page: https://nextjs.org/docs/pages/api-reference/components/head

Can you explain why this change is needed? And the steps to reproduce?

**Comment by PierreRe** [comment]
Correct, the `key` property is only needed when `Head` is used multiple times.

### Example
`Head` may be called in the most basic `components/layout.tsx` to set basic meta-information. `layout.tsx` is repsonsible for rendering the base layout of all pages in the starter example. A common use-case (for me), is to overwrite the meta-information on content-type level (i.e. component-level). This often results in basic pages having different attributes from taxonomy-term pages.
To prevent multiplication of those attributes, `Head` allows using said `key` attribute.

Using the Umami-Demo, `pages/[...slug].tsx` renders `components/layout.tsx` which then renders `components/meta.tsx`. Here, the `meta:image` attribute is provided. It is set to a static image.
As documented, a user may call `Head` again, later in the render process (for example in `component/node--page.tsx`) to set more attributes, or overwrite existing attributes. I observed that resetting the `title` attribute worked as expected. However trying to overwrite the `og:image` resulted in multiplication of the `og:image`-tag in the rendered HTML.

### Reproduce
1. setting a fallback `og:image` in `meta.tsx`
2. setting more correct fallback-image to `og:imge` on `components/node--page.tsx` for nodes
(This should result in duplicate `og_imge`-tags)
3. rendering drupal-paragraphs within `components/node--page.tsx`
4. when rendering a paragraph with a custom flag (for example: "add as social-media image"), re-set `og:image` to it's final image via `Head`

Granted, steps 3 and 4 represent a worst-case scenario, however, step should be quite common.

### Motivation
To give users a guideline on how to (correctly) set and overwrite `Head` attributes using `next/head` I proposed to add a namespaced `key` attribute.

File: examples/example-blog/components/meta.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ export function Meta({ title, tags }: MetaProps) {
   return (
     <Head>
       <link
+        key="head_link_canonical"
```
Comment (JohnAlbin) [inline examples/example-blog/components/meta.tsx:17]: Let's not prefix our key's with `head_` or even with `link_` or `meta_`. It seems overly verbose. If React is trying to dedupe a `link` element, it doesn't need the extra specificity of a prefix that duplicates the HTML structure.

tldr; let's use `key="canonical"` here.

File: examples/example-blog/components/meta.tsx [hunk 2]
```diff
@@ -35,17 +36,19 @@ export function Meta({ title, tags }: MetaProps) {
         })
       ) : (
         <>
-          <title>{`${title} | Next.js for Drupal`}</title>
+          <title key="head_title">{`${title} | Next.js for Drupal`}</title>
           <meta
+            key="head_meta_description"
```
Comment (JohnAlbin) [inline examples/example-blog/components/meta.tsx:41]: `key="description"` please.

**Comment by PierreRe** [comment]
Cheers @JohnAlbin 
You're right, adding a prefix is not necessary, as React dedupes on html elements. I updated this PR to the latest commit in the main branch and removed the unnecessary prefixes. I also went ahead and removed the prefixes on og:image and renamed the key to use underscores to be more in-line with the code-style.
Thanks for your input.

File: examples/example-auth/pages/index.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ export default function IndexPage() {
   return (
     <>
       <Head>
-        <title>Next.js for Drupal | Authentication Example</title>
+        <title key="title">Next.js for Drupal | Authentication Example</title>
```
Comment (JohnAlbin) [inline examples/example-auth/pages/index.tsx:11]: This code example is from the [docs on Next.js that mention keys]():

> To avoid duplicate tags in your head you can use the key property, which will make sure the tag is only rendered once, as in the following example:
> 
> ```js
> function IndexPage() {
>   return (
>     <div>
>       <Head>
>         <title>My page title</title>
>         <meta property="og:title" content="My page title" key="title" />
>       </Head>
>       <Head>
>         <meta property="og:title" content="My new title" key="title" />
>       </Head>
>       <p>Hello world!</p>
>     </div>
>   )
> }
> ```

You can see that don't put a `key` on the title tag.

**Review by JohnAlbin — CHANGES_REQUESTED** [review]
I don't think `<title key="title">` is necessary.

File: examples/example-auth/pages/index.tsx (hunk 3 above)
Comment (PierreRe) [reply examples/example-auth/pages/index.tsx:11]: Agreed, but not for the docs, but for the function behind deduplication of the `<title>` Tag in the actual Head component here: https://github.com/vercel/next.js/blob/c5b5b1e3a3d85d498770a974c250c864e8d2a761/packages/next/src/shared/lib/head.tsx#L84

As per the docs, we could argue that only a single `<title>` is declared and deduplication is therefore not strictly necessarry.
However, the actual NextJS `<Head>` component handles the `<title>` and `<base>` tag differently from the `<meta>` tag. I guess, my initial problem originates in line 106 of the head component which sets the `isUnique` variable to false if a meta-property is not "name"  or has no key.
https://github.com/vercel/next.js/blob/c5b5b1e3a3d85d498770a974c250c864e8d2a761/packages/next/src/shared/lib/head.tsx#L106

It was an interesting dive into the head component,
Cheers

**Review by JohnAlbin — APPROVED** [review]
